src/lib/sticky-header/sticky-header-dir.ts

```diff
@@ -0,0 +1,399 @@
+   * when it is being unstuck
+   */
+  originalCss: any;
+  /**
+   * 'getBoundingClientRect().top' of CdkStickyRegion of current sticky header.
+   * It is used with '_scrollFinish' to judge whether the current header
+   * need to be stuck.
+   */
+  private _containerStart: number;
+  /**
+   * `_scrollFinish` is the place from where the stuck element should be unstuck
+   */
+  private _scrollFinish: number;
+
+  private _onScrollSubscription: Subscription;
+
+  private _onTouchSubscription: Subscription;
+
+  private _onResizeSubscription: Subscription;
+
+  constructor(element: ElementRef,
+              scrollable: Scrollable,
+              @Optional() public parentRegion: CdkStickyRegion,
+              platform: Platform) {
+    if (platform.isBrowser) {
+      this.element = element.nativeElement;
+      this.upperScrollableContainer = scrollable.getElementRef().nativeElement;
+      this.setStrategyAccordingToCompatibility();
+    }
+  }
+
+  ngAfterViewInit(): void {
+    if (!this.isStickyPositionSupported) {
+      this.stickyParent = this.parentRegion != null ?
+        this.parentRegion._elementRef.nativeElement : this.element.parentElement;
+      let values = window.getComputedStyle(this.element, '');
+      this.originalCss = this.generateCssStyle(
+        values.getPropertyValue('zIndex'),
+        values.getPropertyValue('position'),
+        values.getPropertyValue('top'),
+        values.getPropertyValue('right'),
+        values.getPropertyValue('left'),
+        values.getPropertyValue('bottom'),
+        values.getPropertyValue('width'));
+      this.attach();
+      this.defineRestrictionsAndStick();
+    }
+  }
+
+  ngOnDestroy(): void {
+    if (this._onScrollSubscription) {
+      this._onScrollSubscription.unsubscribe();
+    }
+
+    if (this._onResizeSubscription) {
+      this._onResizeSubscription.unsubscribe();
+    }
+
+    if (this._onTouchSubscription) {
+      this._onTouchSubscription.unsubscribe();
+    }
+  }
+
+  /**
+   * getSupportList() is used to get a list of string which can be set to
+   * sticky-header's style.position and make Sticky positioning work.
+   * It returns a list of string.
+   *
+   * According to the "Position:sticky Browser compatibility" in
+   * "https://developer.mozilla.org/en-US/docs/Web/CSS/position".
+   *
+   * For Desktop: Sticky positioning works well on Chrome, Edge, Firefox and Opera. And can
+   * also work well on Safari with a "-webkit-" prefix. It only does not work on IE.
+   *
+   * For Mobile: Sticky positioning works well on Android Webview, Chrome for Android, Edge,
+   * Firefox Mobile, Opera Mobile. And can also work well on Safari Mobile with a "-webkit-" prefix.
+   * It won't always work on IE phone.
+   *
+   * The implementation references the compatibility checking in Modernizer
+   * (https://github.com/Modernizr/Modernizr/blob/master/feature-detects/css/positionsticky.js).
+   */
+  getSupportList(): string[] {
```

Comment:
1. Add the following to `src/cdk/platform/features.ts`:
```ts
let computedPositionStickySupported: boolean | null = null;

/**
 * Whether the browser support css `position: sticky`.
 * Based on the check from modernizr:
 * https://github.com/Modernizr/Modernizr/blob/master/feature-detects/css/positionsticky.js
 */
export function isPositionStickySupported() {
  if (computedPositionStickySupported != null) {
    return computedPositionStickySupported;
  }

  const elementStyle = document.createElement('div').style;
  elementStyle.cssText = ['', '-webkit-'].map(p => `position: ${p}sticky`).join(';');
  computedPositionStickySupported = elementStyle.cssText.indexOf('sticky') !== -1;
  return computedPositionStickySupported;
}
```
(the `ms`, `moz`, and `o` prefixes aren't necessary for us)
2. Consume that function in this component and just always use both the `webkit` and unprefixed styles.